sound/soc/sof/ipc.c

```diff
@@ -366,6 +366,36 @@ void snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id)
 }
 EXPORT_SYMBOL(snd_sof_ipc_reply);
 
+static void ipc_comp_notification(struct snd_sof_dev *sdev,
+				  struct sof_ipc_cmd_hdr *hdr)
+{
+	u32 msg_type = hdr->cmd & SOF_CMD_TYPE_MASK;
+	struct sof_ipc_ctrl_data *cdata;
+	int ret;
+
+	switch (msg_type) {
+	case SOF_IPC_COMP_GET_VALUE:
+	case SOF_IPC_COMP_GET_DATA:
+		cdata = kzalloc(hdr->size, GFP_KERNEL);
+		if (!cdata)
+			return;
+
+		/* read back full message */
+		ret = snd_sof_ipc_msg_data(sdev, NULL, cdata, hdr->size);
+		if (ret < 0) {
+			dev_err(sdev->dev, "failed to read component event: %d\n", ret);
+			return;
+		}
+		break;
+	default:
+		dev_err(sdev->dev, "error: unhandled component message %x\n", msg_type);
```

Comment:
was it `%#x` to include a `0x` prefix?

Reply:
No, it does not.
I have based the skeleton on the ipc_trace_message() / ipc_stream_message(), the `%x` is coming from there.